### PR TITLE
Ruby Project 2: Enumerable Methods

### PR DESCRIPTION
**Pull request by VanessaAoki**
This project aims to re-create the following methods:

1. #each
2. #each_with_index
3. #select
4. #all?
5. #any?
6. #none?
7. count
8. map
9. inject

### CODE REVIEW

File: enumerables.rb [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+      end
+    return false
+  end
+
+  def my_none?
+    my_each do |value|
+      if yield(value) != true
+        return true
+      end
+    return false
```
Comment (MarilenaRoque) [inline enumerables.rb]: - Missing 'end' for the 'each' loop

File: enumerables.rb [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+      end
+    return true
+  end
+  
+  def my_any?
+    my_each do |value|
+      if yield(value) != false
+        return true
+      end
+    return false
```
Comment (MarilenaRoque) [inline enumerables.rb]: Missing 'end' for the 'each' loop

File: enumerables.rb [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+      if my_proc.nil?
+        arr.push(my_proc.call(value))        
+      else
+        arr.push(yield(value))
+      end
+    end
+    return arr
+  end
+
+  def my_inject(inject=self(0))
```
Comment (MarilenaRoque) [inline enumerables.rb]: - Self(0) is raising an error. Should it be Self[0]?

File: enumerables.rb [hunk 4]
```diff
@@ -0,0 +1,92 @@
...
+      end
+    i
+  end
+
+  def my_all?
+    my_each do |value|
+      if yield(value) != true
+        return false
+      end
+    return true
```
Comment (MarilenaRoque) [inline enumerables.rb]: - Missing 'end' for the 'each' loop

**Review by MarilenaRoque — CHANGES_REQUESTED** [review]
# Hello @VanessaAoki! :rocket:

### Good Job on your project, so far!
Now, let's correct a few errors and adjust the behavior of your function in order to complete that. 😃 

## Changes Required ♻️ 

- [x] Your code is raising some errors, please check my comments below and make sure that you can run your 'enumerables.rb' flawlessly.
- [x] Your methods should be able to receive the same arguments and should present the same behavior as the original enumerable methods of ruby.
- [x] A file to call and test your functions is not required, in order to avoid the introduction of bugs, I suggest you add your test.rb in your .gitignore and use it only locally.


**To fix your methods behavior:**  Please check the documentation for the original Enumerable methods [here](https://ruby-doc.org/core-3.0.0/Enumerable.html). You will see that after the explanation there is a box with examples with different inputs for the method. Make sure that your method also covers all these types of input.

Feel free to message me on Slack, if you need further info about the review.
Happy Coding!!! 💻 

**_Marilena Roque
Full-Stack Developer
Code Reviewer at Microverse Inc.
[Portfolio](https://mariroqueproject.com/) | [Linkedin](https://www.linkedin.com/in/roquemarilena/) 
marilenas@id.uf.br_**

**Review by MahmoudBakr23 — APPROVED** [review]
# **APPROVED ✔✔**

<hr>

Hello @VanessaAoki, Well done, and congratulations for completing this project to meet its specifications and also for doing the requested changes 🎉🎉👍👍

You can now merge and submit, 👍👍 but also you can take 10 mins to do the following OPTIONAL point for easier access to the project. 🙏🙏

- Kindly you could deploy the project on a live demo link i.e. repl.it and create some examples for each Enumerable method you created in order to run directly on the live demo link

<hr>

Good luck with the next and happy coding 👨‍💻💻

**Pull request merged**